### DISCUSSION TITLE
How to add multiple path in PMD

### DISCUSSION THREAD

**Discussion by mohdrashid1** [pmd/pmd#2571] [Q&A] [answered]
**Description:**

/home/mohd/findbugs-3.0.1/bin/run.sh pmd -d /home/mohd/Desktop/Fxt/Codegen/src/main/java /home/mohd/Desktop/Fxt/VC-Git-Skill/src/main/java -f xml -failOnViolation false -reportfile pmd.xml -R rulesets/java/quickstart.xml

When i execute this command i got error 
Was passed main parameter '/var/jenkins_home/workspace/Fx-VC-Git-Skill-Bot/Codegen/src/main/java' but no main parameter was defined in your arg class
Build step 'Execute shell' marked build as failure
Finished: FAILURE

Please help how to add multiple path in PMD

**Comment by adangel** [comment, marked as answer]
Specifying multiple paths is not supported unfortunately. Here's what's supported: https://pmd.github.io/latest/pmd_userdocs_cli_reference.html

You can do the following:
* Specify the common base directory `-d /home/mohd/Desktop/Fxt/` - then your two projects will be included, but maybe more projects, that you don't want to have analyzed.
* Create a filelist for the files, e.g. `find /home/mohd/Desktop/Fxt/Codegen/src/main/java /home/mohd/Desktop/Fxt/VC-Git-Skill/src/main/java -name "*.java" > filelist.txt` and use that via `-filelist filelist.txt`

**Comment by mohdrashid1** [comment]
@adangel Thanks for your comment but i found the solution its work for me

**Reply by iwamfer** [reply]
Hi @mohdrashid1 , please could you share what solution worked for you? 
I can't use the solution of common base directory because there are files in same base directory that I can't run PMD. Out of scope. 

Tks
